plugins/ipam/host-local/config.go

```diff
@@ -39,31 +39,29 @@ type IPAMArgs struct {
 }
 
 type Net struct {
-	Name string      `json:"name"`
-	IPAM *IPAMConfig `json:"ipam"`
+	Name  string      `json:"name"`
+	IPAM  *IPAMConfig `json:"ipam"`
+	IPAM6 *IPAMConfig `json:"ipam6"`
 }
 
 // NewIPAMConfig creates a NetworkConfig from the given network name.
-func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, error) {
+func LoadIPAMConfig(bytes []byte, args string) (*IPAMConfig, *IPAMConfig, error) {
 	n := Net{}
 	if err := json.Unmarshal(bytes, &n); err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	if args != "" {
 		ipamArgs := IPAMArgs{}
 		err := types.LoadArgs(args, &ipamArgs)
 		if err != nil {
-			return nil, err
+			return nil, nil, err
 		}
 	}
 
-	if n.IPAM == nil {
-		return nil, fmt.Errorf("%q missing 'ipam' key")
+	if n.IPAM == nil && n.IPAM6 == nil {
+		return nil, nil, fmt.Errorf("Need at least one of ipam or ipam6 key")
 	}
 
-	// Copy net name into IPAM so not to drag Net struct around
-	n.IPAM.Name = n.Name
```

Comment:
Also, the Name field of IPAMConfig doesn't get marshalled at all, so it's going to be "" in both structures now.  And that now gets passed directly to disk.New() and will be the same for both IPv4 and IPv6.  First, I think each IPAM struct needs a name, so maybe just make n.IPAM.Name = n.Name, and n.IPAM6.Name = n.Name + "6"?  That way the stores aren't confused either.